pkg/services/searchV2/auth.go

```diff
@@ -1,14 +1,17 @@
 package searchV2
 
 import (
-	"fmt"
+	"context"
 
 	"github.com/grafana/grafana/pkg/models"
 	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/services/sqlstore/permissions"
 )
 
 // ResourceFilter checks if a given a uid (resource identifier) check if we have the requested permission
-type ResourceFilter func(uid string) bool
+type ResourceFilter interface {
+	check(uid string) bool
```

Comment:
any reason this is better than having a function?

It does let us put the lookup map in a struct... but I think we could do the same in a closure